bigquery/google/cloud/bigquery/job.py

```diff
@@ -719,9 +719,11 @@ class _JobConfig(object):
         job_type (str): The key to use for the job configuration.
     """
 
-    def __init__(self, job_type):
+    def __init__(self, job_type, **properties):
```

Comment:
Let's rename to `**kwargs` to follow that convention.

See:

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google

Regarding the docs, let's say something like:

```
Set properties on the constructed configuration by using the property
name as the name of a keyword argument.
```

on all the subclasses.